org.osgi.test.junit5/src/main/java/org/osgi/test/junit5/config/ConfigurationExtension.java

```diff
@@ -0,0 +1,156 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.junit5.config;
+
+import static org.osgi.test.common.inject.FieldInjector.findAnnotatedFields;
+import static org.osgi.test.common.inject.FieldInjector.setField;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.lang.reflect.Parameter;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionConfigurationException;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.test.common.annotation.config.Configuration;
+import org.osgi.test.common.annotation.config.Configurations;
+import org.osgi.test.common.dictionary.Dictionaries;
+
+public class ConfigurationExtension implements BeforeEachCallback, ParameterResolver {
+
+	final static Namespace NAMESPACE = Namespace.create(ConfigurationExtension.class);
+
+	@Override
+	public void beforeEach(ExtensionContext extensionContext) throws Exception {
+
+		List<Field> fields = findAnnotatedFields(extensionContext.getRequiredTestClass(), Configuration.class);
+
+		BundleContext bundleContext = FrameworkUtil.getBundle(extensionContext.getRequiredTestClass())
+			.getBundleContext();
+
+		fields.forEach(field -> {
+			assertValidFieldCandidate(field);
+
+			Configuration configAnnotation = field.getAnnotation(Configuration.class);
+			Class<?> memberType = field.getType();
+			Type genericMemberType = field.getGenericType();
+
+			setField(field, extensionContext.getRequiredTestInstance(),
+				resolveReturnValue(configAnnotation, bundleContext, extensionContext));
+		});
+
+		extensionContext.getElement()
+			.ifPresent((element) -> {
+				Configurations configAnnotations = element.getAnnotation(Configurations.class);
+				if (configAnnotations != null) {
+					Stream.of(configAnnotations.value())
+						.forEachOrdered((configAnnotation) -> {
+							resolveReturnValue(configAnnotation, bundleContext, extensionContext);
+						});
+				}
+				Configuration configAnnotation = element.getAnnotation(Configuration.class);
+				if (configAnnotation != null) {
+					resolveReturnValue(configAnnotation, bundleContext, extensionContext);
+				}
+			});
+	}
+
+	private static org.osgi.service.cm.Configuration resolveReturnValue(Configuration configurationAnnotation,
```

Comment:
If you follow my other suggestion and use `@InjectConfiguration` instead of `@Configuration` for the annotation, then that will eliminate the name collision and you won't need to use the FQN, which would make this a bit cleaner. I think generally it will be common for users of this code to want to refer to both so it will make all the users of this code cleaner too.